pkg/client/unversioned/extensions.go

```diff
@@ -94,6 +96,10 @@ func (c *ExtensionsClient) ThirdPartyResources(namespace string) ThirdPartyResou
 	return newThirdPartyResources(c, namespace)
 }
 
+func (c *ExtensionsClient) Generic(namespace string, gvk unversioned.GroupVersionKind) GenericResourceInterface {
+	return newGenericResources(c, namespace, gvk)
```

Comment:
This doesn't really belong to the ExtensionsClient. IMO, it should be parallel to the dynamic client krousey mentioned.

For now, I think it makes more sense to create a GenericClient, parallel to ExtensionsClient and Client, and it should be constructed with a RESTClient. WDYT?